### PR TITLE
feat: Expose message event details for Iframe Widget

### LINKED ISSUE #13419: [Feature] Message event details
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Implement a newer version of the IframeWidget, so that users can use the newer version and have access to all the metadata provided by the [`message`](https://developer.mozilla.org/en-US/docs/Web/API/Window/message_event) event.

### Why should this be worked on?

The browser's [`MessageEvent` ](https://developer.mozilla.org/en-US/docs/Web/API/Window/message_event)provides a few metadata which can be useful when building web application checks and workflows.
In the IframeWidget, we only have access to the event's data, but not the other properties like `source`, `origin`, etc.

### PR DESCRIPTION
**Pull request by souma-ghosh**
## Description
In Iframe widget some [MessageEvent](https://developer.mozilla.org/en-US/docs/Web/API/Window/message_event) metadata are not available to the Appsmith developer which might be useful when building web application checks and workflows.
This PR exposes the metadata related to MessageEvent in messageEventProps key for Iframe Widget.
Also enables messageEventProps key in autosuggest.

Fixes #13419

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?

- Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/IframeWidget/widget/index.tsx [hunk 1]
```diff
@@ -186,8 +187,18 @@ class IframeWidget extends BaseWidget<IframeWidgetProps, WidgetState> {
...
-  handleMessageReceive = (event: MessageEvent) => {
-    this.props.updateWidgetMetaProperty("message", event.data, {
+  handleMessageReceive = ({
+    data,
+    lastEventId,
+    origin,
+    ports,
+  }: MessageEvent) => {
+    this.props.updateWidgetMetaProperty("messageEventProps", {
+      lastEventId,
```
Comment (ashit-rath) [inline app/client/src/widgets/IframeWidget/widget/index.tsx:197]: Are we getting issues when we add `source` as well to the meta property?

File: app/client/src/widgets/IframeWidget/widget/index.tsx [hunk 2]
```diff
@@ -159,6 +159,7 @@ class IframeWidget extends BaseWidget<IframeWidgetProps, WidgetState> {
   static getMetaPropertiesMap(): Record<string, any> {
     return {
       message: undefined,
+      messageEventProps: undefined,
```
Comment (ashit-rath) [inline app/client/src/widgets/IframeWidget/widget/index.tsx:162]: Shall we name this property as `messageMetadata`? The word `Props` is very React-ish and `Metadata` is a generic term for such things.

**Review by ashit-rath — COMMENTED** [review]
Can we add some cypress test for this?
It could be a simple IFrame with a button in it that posts a simple message on click
To verify the data received, we could have a Text widget and bind the Iframe metadata property.

Let me know if this makes sense.

File: app/client/src/widgets/IframeWidget/widget/index.tsx (hunk 2 above)
Comment (souma-ghosh) [reply app/client/src/widgets/IframeWidget/widget/index.tsx:162]: Sure will update this

**Comment by souma-ghosh** [comment]
> Can we add some cypress test for this? It could be a simple IFrame with a button in it that posts a simple message on click To verify the data received, we could have a Text widget and bind the Iframe metadata property.
> 
> Let me know if this makes sense.

@ashit-rath Here we will run into a blocker. We won't be able to access the button in the Iframe through Cypress code. There is a workaround for this though where we will need to turn off `chromeWebsiteSecurity` in our Cypress config as mentioned in [this](https://www.cypress.io/blog/2020/02/12/working-with-iframes-in-cypress/) article. Please let me know your thoughts about this.

**Comment by souma-ghosh** [comment]
@ashit-rath Tried with the srcDoc this time. Worked! Also referred to the article again, the Cypress test is working fine now.

**Review by sbalaji1192 — APPROVED** [review]

**Comment by vivekverma2312** [comment]
@souma-ghosh Tested and LGTM.

**Comment by somangshu** [comment]
@Aishwarya-U-R I merged this by mistake, Please let me know if we need a revert here!

**Pull request merged**